lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -352,6 +352,13 @@ impl<'s> SegmentHolder {
                 }
             }
 
+            if applied_points.contains(&point_id) {
+                // point was already applied to another segment
+                // we can remove the duplicate and continue
+                write_segment.delete_point(op_num, point_id)?;
+                return Ok(false);
+            }
```

Comment:
This is not related to snapshots directly, but I though it might be good optimization. This prevents us from double-applying operations.
Not sure if there is a scenario where it could happen, though. So happy to remove it if anybody objects.